src/rpc/Errors.h

```diff
@@ -0,0 +1,206 @@
+#ifndef REPORTING_RPC_ERRORS_H_INCLUDED
+#define REPORTING_RPC_ERRORS_H_INCLUDED
+
+#include <ripple/protocol/ErrorCodes.h>
+
+#include <boost/json.hpp>
+
+#include <optional>
+#include <string>
+#include <string_view>
+#include <variant>
+
+namespace RPC {
+
+/**
+ * @brief Custom clio RPC Errors.
+ */
+enum class ClioError {
+    rpcMALFORMED_CURRENCY = 5000,
+};
+
+/**
+ * @brief Holds info about a particular @ref ClioError.
+ */
+struct ClioErrorInfo
+{
+    ClioError const code;
+    std::string_view const error;
+    std::string_view const message;
+};
+
+/**
+ * @brief Clio uses compatible Rippled error codes for most RPC errors.
+ */
+using Error = ripple::error_code_i;
```

Comment:
Should we call this `RippledError`?

Reply:
We can. I left it as `Error` because it's less changes on call sites - we use this `Error` quite a bit. Let me try to rename it

Reply:
oh thats true, lots of changes, we could make a separate issue/PR and do that later

Reply:
In the same branch, after changing `Error` to `RippledError` i get `28 files changed, 289 insertions(+), 226 deletions(-)
`. WDYT should we just do it as part of this PR then?

Reply:
thats not too bad, I'm ok with that if you are, but don't feel strongly

Reply:
Let's just get it done :+1: